templates/_macros/quote-wrapper.jinja

```diff
@@ -0,0 +1,167 @@
+      {%- macro _citation_block_source_name() -%}
+        <!--Optional citation source name-->
+        <h5 class="u-no-margin--bottom u-no-padding--top">{{ citation_source_name }}</h5>
+      {%- endmacro -%}
+      {%- macro _citation_block_source_title() -%}
+        <!--Optional citation source title-->
+        <p class="u-text--muted u-no-padding u-no-margin">{{ citation_source_title }}</p>
+      {%- endmacro -%}
+      <div class="p-section--shallow">
+        {%- if has_citation_source_name and has_citation_source_title %}
+          <!-- When source name and title are both present, wrap them in a 50/50 row to split them on medium -->
+          <div class="row">
+            <div class="col-3 col-medium-3">
+              {{ _citation_block_source_name() }}
+            </div>
+            <div class="col-3 col-medium-3">
+              {{ _citation_block_source_title() }}
+            </div>
+          </div>
+        {%- else -%}
+          {%- if has_citation_source_name %}
+            <!--Optional citation source name-->
+            {{ _citation_block_source_name() }}
+          {%- endif -%}
+          {%- if has_citation_source_title %}
+            <!--Optional citation source title-->
+            {{ _citation_block_source_title() }}
+          {%- endif -%}
+        {%- endif -%}
+      </div>
+    {%- endif -%}
+  {%- endmacro %}
+
+  {% if has_heading_row %}
+    <!--Optional heading -->
+    <hr class="p-rule--highlight">
+    <div class="row">
+      {%- if has_title %}
+        <!--Optional heading text-->
+        <div class="col-3 col-medium-3">
+          <p class="p-muted-heading">{{ title }}</p>
+        </div>
+      {%- endif -%}
+
+      {%- if has_heading_link %}
+        <!--Optional heading link -->
+        <div class="col-3 col-medium-3 col-start-large-10 col-start-medium-5">
+          <p class="p-text--default">
+           {{ heading_link_content }}
+          </p>
+        </div>
+      {% endif -%}
+    </div>
+  {% endif -%}
+  <div class="row">
+    {% if has_signpost_image -%}
+      <!--Optional signpost image-->
+      <div class="col-3">
+        <div class="row">
+          <div class="col-2">
+            <div class="p-section--shallow">
+              {#- TODO find a better way to line the image top up with the quote and citation -#}
+              <div class="u-hide--medium u-hide--small u-sv1"></div>
+              {{ signpost_image_content }}
+            </div>
+          </div>
+        </div>
+      </div>
+    {% endif -%}
+    {#- Skip to fourth column if the signpost_image (which takes up first three columns) is missing -#}
+    <div class="col-9{% if not has_signpost_image %} col-start-large-4{% endif %}">
+      <div class="u-hide--medium u-hide--small">
+        <hr>
+        {#- TODO find a better way to space the quote and citation away from the rule -#}
+        <div class="u-sv1"></div>
+      </div>
```

Comment:
@pastelcyborg regarding my [previous comment](https://github.com/canonical/vanilla-framework/pull/5275#issuecomment-2289130361), here's the strangeness around spacing text away from the rule. 

The text of the quote and citation source name are aligned vertically because they all have no top padding. Adding a rule above them as-is makes the rule far too close to the text, so I use vertical spacing utility to compensate. Is this the sort of thing that will require additional scss?

Reply:
I think I had some weird cache weirdness going on when I reported this, as previously `hr` was getting no bottom margin, but now it is spacing the quote and citation out correctly as long as I don't use `p-rule`. So this is tangentially related to #5241 , but is not blocked by it.